acme/acme/standalone_test.py

```diff
@@ -264,34 +265,36 @@ def setUp(self):
                     os.path.join(localhost_dir, 'key.pem'))
 
         from acme.standalone import simple_tls_sni_01_server
-        self.thread = threading.Thread(
-            target=simple_tls_sni_01_server, kwargs={
-                'cli_args': ('filename',),
-                'forever': False,
-            },
-        )
+        self.port = 40000
```

Comment:
I don't think we should hardcode a port like this.

As a more minor point, why are these changes in this PR?

Reply:
In fact this code was subject to deadlock, and it was already the case on Windows. It started also to be it in this PR. So I refactor to not have this kind of deadlock.

For the port, I would argue that this kind of hardcoded port is in several places in certbot tests: I am not talking of parameters in mocked objects, but in integration tests with real bindings on the system that execute them.

Reply:
> In fact this code was subject to deadlock, and it was already the case on Windows. It started also to be it in this PR on Linux.

I don't see how the other changes in this PR cause a deadlock on Linux here.

> For the port, I would argue that this kind of hardcoded port is in several places in certbot tests: I am not talking of parameters in mocked objects, but in integration tests with real bindings on the system that execute them.

I think the major difference is our unit tests are shipped in the Python package and usually run by OS packagers to see if the code is working correctly on their system. We don't have the same control on these environments as we do as on the environment for our integration tests so I think we should try and make our unit tests more robust and avoid doing things like hardcoding ports.

Reply:
I do not see neither why it suddenly deadlocked. But since it already happened, in similar code, I preferred to rewrite.

If I solve the static port attribution, to get randomly an available one, like in the previous implementation, will you be OK with the modifications I did here?

Reply:
> If I solve the static port attribution, to get randomly an available one, like in the previous implementation, will you be OK with the modifications I did here?

I'll need to review the test changes more closely when it's finished, but I should be!

Reply:
Done. I pick a random free port.